subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppBasePlugin.java

```diff
@@ -75,12 +74,7 @@ public void apply(final ProjectInternal project) {
         final ProviderFactory providers = project.getProviders();
 
         // Enable the use of Gradle metadata. This is a temporary opt-in switch until available by default
-        project.getRepositories().withType(ComponentAwareRepository.class, new Action<ComponentAwareRepository>() {
-            @Override
-            public void execute(ComponentAwareRepository componentAwareRepository) {
-                componentAwareRepository.useGradleMetadata();
-            }
-        });
+        project.getGradle().getExperimentalFeatures().enableAll();
```

Comment:
:o: I'm wondering if this should be _per project_. Imagine a polyglot project with one CPP module and a Java one. Nothing critical here, but something to think about.

Reply:
The entire Gradle execution either has experimental features enabled or not. I want to avoid providing fine-grained control over experimental features, unless there's a good use case.